### PR TITLE
Expose the underlying Node's Mark on Decoder

### PR DESCRIPTION
**Pull request by brentleyjones**
This allows accessing source location during decoding.

### CODE REVIEW

File: Sources/Yams/Decoder.swift [hunk 1]
```diff
@@ -344,6 +344,19 @@ extension URL: ScalarConstructible {
     }
 }
 
+// MARK: Decoder.node
```
Comment (jpsim) [inline Sources/Yams/Decoder.swift:347]: Nit:
```suggestion
// MARK: Decoder.mark
```

File: Sources/Yams/Decoder.swift [hunk 2]
```diff
@@ -344,6 +344,19 @@ extension URL: ScalarConstructible {
...
+
+extension Decoder {
+    /// The `Mark` for the underlying `Node` that has been decoded.
+    public var mark: Mark? {
+        guard let decoder = self as? _Decoder else {
+            return nil
+        }
+
+        return decoder.node.mark
```
Comment (jpsim) [inline Sources/Yams/Decoder.swift:356]: Nit:

```suggestion
        return (self as? _Decoder)?.node.mark
```

**Review by jpsim — COMMENTED** [review]
Seems like a very useful thing to have access to, thanks for the PR!

I just have two small nitpicky suggestions. Could you also please add a test and a changelog entry?

File: Sources/Yams/Decoder.swift (hunk 1 above)
Comment (brentleyjones) [reply Sources/Yams/Decoder.swift:347]: :+1: Missed in a refactor.

File: Tests/YamsTests/EncoderTests.swift [hunk 3]
```diff
@@ -348,6 +348,26 @@ class EncoderTests: XCTestCase { // swiftlint:disable:this type_body_length
...
+            init(from decoder: Decoder) throws {
+                let container =  try decoder.singleValueContainer()
+
+                self.value = try container.decode(String.self)
+                self.location = decoder.mark?.description ?? ""
+            }
+        }
+        let t = try YAMLDecoder().decode(Test.self, from: s)
+        XCTAssertEqual(t.value, "Value")
+        XCTAssertEqual(t.location, "1:1")
```
Comment (brentleyjones) [inline Tests/YamsTests/EncoderTests.swift:368]: I would have tested a mapping here, but #297 prevents the value from being what it should be.

Comment (jpsim) [reply Tests/YamsTests/EncoderTests.swift:368]: Can you rebase this PR after merging https://github.com/jpsim/Yams/pull/297 so you can add that test?

Comment (brentleyjones) [reply Tests/YamsTests/EncoderTests.swift:368]: :+1: I'll also add the linux change needed.

**Review by jpsim — APPROVED** [review]

**Pull request merged**